GHC 8.6.1 support

Master branch doesn't build on ghc 8.6.1 (tested with stack on resolver `nightly-2018-10-04`):

```
/XXX/libmpd-haskell/src/Network/MPD/Core.hs:120:13: error:
    • No instance for (Control.Monad.Fail.MonadFail MPD)
        arising from a do statement
        with the failable pattern ‘[msg]’
    • In a stmt of a 'do' block: [msg] <- send ""
      In the expression:    
        do [msg] <- send "" 
           if "OK MPD" `isPrefixOf` msg then
               MPD $ checkVersion $ parseVersion msg
           else             
               return False 
      In an equation for ‘checkConn’:
          checkConn         
            = do [msg] <- send ""
                 if "OK MPD" `isPrefixOf` msg then
                     MPD $ checkVersion $ parseVersion msg
                 else       
                     return False
    |                       
120 |             [msg] <- send ""
    | 
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -49,6 +49,10 @@ matrix:
     compiler: ": #GHC 8.4.1"
     addons: {apt: {packages: [cabal-install-1.24,ghc-8.4.1,happy-1.19.5,alex-3.1.7], sources: [hvr-ghc]}}
 
+  - env: BUILD=cabal GHCVER=8.4.1 CABALVER=1.24 HAPPYVER=1.19.5 ALEXVER=3.1.7
+    compiler: ": #GHC 8.6.1"
+    addons: {apt: {packages: [cabal-install-1.24,ghc-8.6.1,happy-1.19.5,alex-3.1.7], sources: [hvr-ghc]}}
+
   - env: BUILD=cabal GHCVER=head  CABALVER=head HAPPYVER=1.19.5 ALEXVER=3.1.7
     compiler: ": #GHC HEAD"
     addons: {apt: {packages: [cabal-install-head,ghc-head,happy-1.19.5,alex-3.1.7], sources: [hvr-ghc]}}
@@ -72,6 +76,10 @@ matrix:
 
   - env: BUILD=stack ARGS="--resolver lts-10"
     compiler: ": #stack 8.2.2"
+    addons: {apt: {packages: [ghc-8.2.2], sources: [hvr-ghc]}}
+
+  - env: BUILD=stack ARGS="--resolver nightly-2018-10-04"
+    compiler: ": #stack 8.6.1 2018-10-04"
     addons: {apt: {packages: [ghc-8.2.2], sources: [hvr-ghc]}}
 
   - env: BUILD=stack ARGS="--resolver nightly"

--- a/.travis.yml
+++ b/.travis.yml
@@ -49,7 +49,7 @@ matrix:
     compiler: ": #GHC 8.4.1"
     addons: {apt: {packages: [cabal-install-1.24,ghc-8.4.1,happy-1.19.5,alex-3.1.7], sources: [hvr-ghc]}}
 
-  - env: BUILD=cabal GHCVER=8.4.1 CABALVER=1.24 HAPPYVER=1.19.5 ALEXVER=3.1.7
+  - env: BUILD=cabal GHCVER=8.6.1 CABALVER=1.24 HAPPYVER=1.19.5 ALEXVER=3.1.7
     compiler: ": #GHC 8.6.1"
     addons: {apt: {packages: [cabal-install-1.24,ghc-8.6.1,happy-1.19.5,alex-3.1.7], sources: [hvr-ghc]}}
 
@@ -80,7 +80,7 @@ matrix:
 
   - env: BUILD=stack ARGS="--resolver nightly-2018-10-04"
     compiler: ": #stack 8.6.1 2018-10-04"
-    addons: {apt: {packages: [ghc-8.2.2], sources: [hvr-ghc]}}
+    addons: {apt: {packages: [ghc-8.6.1], sources: [hvr-ghc]}}
 
   - env: BUILD=stack ARGS="--resolver nightly"
     compiler: ": #stack nightly"

--- a/src/Network/MPD/Commands/Types.hs
+++ b/src/Network/MPD/Commands/Types.hs
@@ -44,6 +44,7 @@ import           Network.MPD.Commands.Arg (MPDArg(prep), Args(Args))
 import           Data.Default.Class
 
 import qualified Data.Map as M
+import           Data.Map.Strict (insertWith)
 import           Data.Time.Clock (UTCTime)
 import           Data.String
 
@@ -263,7 +264,7 @@ sgGetTag meta s = M.lookup meta $ sgTags s
 
 -- | Add metadata tag value.
 sgAddTag :: Metadata -> Value -> Song -> Song
-sgAddTag meta value s = s { sgTags = M.insertWith' (++) meta [value] (sgTags s) }
+sgAddTag meta value s = s { sgTags = insertWith (++) meta [value] (sgTags s) }
 
 defaultSong :: Path -> Song
 defaultSong path =

--- a/src/Network/MPD/Core.hs
+++ b/src/Network/MPD/Core.hs
@@ -117,7 +117,8 @@ mpdOpen = MPD $ do
             (Just <$> connectTo host (PortNumber $ fromInteger port))
             `catchAny` const (return Nothing)
         checkConn = do
-            [msg] <- send ""
+            singleMsg <- send ""
+            let [msg] = singleMsg
             if "OK MPD" `isPrefixOf` msg
                 then MPD $ checkVersion $ parseVersion msg
                 else return False

